integration/spark2/src/main/commonTo2.2And2.3/org/apache/spark/sql/hive/CarbonAnalyzer.scala

```diff
@@ -40,12 +40,13 @@ class CarbonAnalyzer(catalog: SessionCatalog,
 
   override def execute(plan: LogicalPlan): LogicalPlan = {
     var logicalPlan = analyzer.execute(plan)
-    logicalPlan = CarbonPreAggregateDataLoadingRules(sparkSession).apply(logicalPlan)
-    logicalPlan = CarbonPreAggregateQueryRules(sparkSession).apply(logicalPlan)
-    if (mvPlan != null) {
-      mvPlan.apply(logicalPlan)
-    } else {
-      logicalPlan
+    if (!logicalPlan.isInstanceOf[Command] && !logicalPlan.isInstanceOf[DeserializeToObject]) {
```

Comment:
Why it needs to be added here. This check already exist in `MVAnalyzerRule.isValidPlan`.  Please add testcase why we need check here.

Reply:
un-relevant plan does not need to apply anything of mv

Reply:
@ravipesala i think, we need to add here, because isValidPlan comes after initializing datamap catalogs, so if no catalog found, it will create catalog and try to register schema, which will go to MV plan flow. so better to avoid this, can you please check once

Reply:
@akashrn5 I don't get the problem, why it is an issue if it initializes for the first time. This is the generic rule we should not feature specific checks here.

Reply:
@ravipesala what i meant is, for example consider use database_name. There is no need of applying these rules for this command right. simply we go through that flow. So it is better to avoid right?

Reply:
It is already avoided inside MVAnalyzerRule, why need to avoid in generic rule?